src/Stratis.Bitcoin.Tests/Consensus/ChainedHeaderTreeTest.cs

```diff
@@ -196,6 +208,46 @@ public void ConnectHeaders_SupplyHeadersThenSupplyMore_Both_Tip_PeerId_Maps_Shou
             Assert.True(peerTipsByPeerIdBefore.Values.Count == peerTipsByPeerIdAfter.Values.Count);
         }
 
+        /// <summary>
+        /// Issue 5 @ Create chained header tree component #1321
+        /// 2 peers should supply headers - one of the peers creates a fork.
+        /// Make sure that ChainedBlock is not created more than once for the same header.
+        /// Check that next pointers were created correctly.
+        /// </summary>
+        [Fact]
+        public void ConnectHeaders_HeadersFromTwoPeersWithFork_ShouldCreateBlocksForNewHeaders()
+        {
+            // Setup
+            var ctx = new TestContext();
+            ChainedHeaderTree cht = ctx.CreateChainedHeaderTree();
```

Comment:
`var chainedHeaderTree = testContext.CreateChainedHeaderTree();`  would be easier to read.